src/Chapter3.hs

```diff
@@ -907,9 +1048,26 @@ Implement instances of "Append" for the following types:
   ✧ *(Challenge): "Maybe" where append is appending of values inside "Just" constructors
 
 -}
+
 class Append a where
     append :: a -> a -> a
 
+newtype Golden = Golden Int -- Golden instead of gold because Gold has been used already
+
+instance Append Golden where
+    append :: Golden -> Golden -> Golden
+    append (Golden a) (Golden b) = Golden(a + b)
+
+instance Append [a] where
+    append :: [a] -> [a] -> [a]
+    append a b = a ++ b
+
+instance Append a => Append (Maybe a) where
+    append :: (Maybe a) -> (Maybe a) -> (Maybe a)
+    append Nothing Nothing = Nothing
+    append a Nothing = a
+    append Nothing a = a
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂